### PR TITLE
All the basic features

### PR DESCRIPTION
**Pull request by timyates**
The first 4 items on the main page

### CODE REVIEW

**Comment by timyates** [comment]
Works in intellij, fails in gradle 🙄

File: Micronaut/src/main/kotlin/io/micronaut/turbo/basic/NonExistentController.kt [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+import io.micronaut.http.annotation.Controller
+import io.micronaut.http.annotation.Get
+import io.micronaut.http.annotation.Status
+
+@Controller("/nonexistent")
+class NonExistentController {
+
+    @Status(HttpStatus.NOT_FOUND)
+    @Get(produces = [MediaType.TEXT_HTML], consumes = [MediaType.TEXT_HTML])
+    fun index() = "Not Found"
```
Comment (sdelamo) [inline Micronaut/src/main/kotlin/io/micronaut/turbo/basic/NonExistentController.kt:14]: Is this missing the `@View` annotation. We say we are returning HTML wit the content-type but it seems to return just a string.

File: Micronaut/src/main/kotlin/io/micronaut/turbo/basic/OneController.kt [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+import io.micronaut.http.annotation.Get
+import io.micronaut.turbo.model.ViewModel
+import io.micronaut.views.View
+import io.micronaut.views.turbo.TurboView
+
+@Controller("/one")
+class OneController {
+
+    @View("one")
+    @TurboView("one")
```
Comment (sdelamo) [inline Micronaut/src/main/kotlin/io/micronaut/turbo/basic/OneController.kt]: remove `@TurboView`?

File: Micronaut/src/main/kotlin/io/micronaut/turbo/RootController.kt [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+import io.micronaut.http.annotation.Get
+import io.micronaut.turbo.model.ViewModel
+import io.micronaut.views.View
+import io.micronaut.views.turbo.TurboView
+
+@Controller
+class RootController {
+
+    @View("index")
+    @TurboView("index")
```
Comment (sdelamo) [inline Micronaut/src/main/kotlin/io/micronaut/turbo/RootController.kt]: remove `@TurboView`?

File: Micronaut/src/main/kotlin/io/micronaut/turbo/basic/LongController.kt [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+import io.micronaut.http.annotation.Get
+import io.micronaut.turbo.model.ViewModel
+import io.micronaut.views.View
+import io.micronaut.views.turbo.TurboView
+
+@Controller("/long")
+class LongController {
+
+    @View("long")
+    @TurboView("long")
```
Comment (sdelamo) [inline Micronaut/src/main/kotlin/io/micronaut/turbo/basic/LongController.kt]: remove `@TurboView`?

File: Micronaut/src/main/kotlin/io/micronaut/turbo/basic/SlowController.kt [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+import io.micronaut.turbo.model.ViewModel
+import io.micronaut.views.View
+import io.micronaut.views.turbo.TurboView
+import java.util.concurrent.CompletableFuture
+
+@Controller("/slow")
+class SlowController {
+
+    @View("slow")
+    @TurboView("slow")
```
Comment (sdelamo) [inline Micronaut/src/main/kotlin/io/micronaut/turbo/basic/SlowController.kt]: remove `@TurboView`?

File: Micronaut/src/main/kotlin/io/micronaut/turbo/basic/ScrollController.kt [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+import io.micronaut.http.annotation.Get
+import io.micronaut.turbo.model.ViewModel
+import io.micronaut.views.View
+import io.micronaut.views.turbo.TurboView
+
+@Controller("/scroll")
+class ScrollController {
+
+    @View("scroll")
+    @TurboView("scroll")
```
Comment (sdelamo) [inline Micronaut/src/main/kotlin/io/micronaut/turbo/basic/ScrollController.kt]: remove `@TurboView`?

File: Micronaut/src/main/kotlin/io/micronaut/turbo/basic/TwoController.kt [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+import io.micronaut.http.annotation.QueryValue
+import io.micronaut.turbo.model.ViewModel
+import io.micronaut.views.View
+import io.micronaut.views.turbo.TurboView
+
+@Controller("/two")
+class TwoController {
+
+    @View("two")
+    @TurboView("two")
```
Comment (sdelamo) [inline Micronaut/src/main/kotlin/io/micronaut/turbo/basic/TwoController.kt]: remove `@TurboView`?

File: Micronaut/src/main/resources/assets/turbolinks-5.2.0.js [hunk 8]
```diff
@@ -1,6 +0,0 @@
...
...wrap(window.location),this.restorationIdentifier=e.uuid(),this.history.start(),this.history.replace(this.location,this.restorationIdentifier)},r.prototype.stopHistory=function(){return this.history.stop()},r.prototype.pushHistoryWithLocationAndRestorationIdentifier=function(t,r){return this.restorationIdentifier=r,this.location=e.Location.wrap(t),this.history.push(this.location,this.restorationIdentifier)},r.prototype.replaceHistoryWithLocationAndRestorationIdentifier=function(t,r){return this.restorationIdentifier=r,this.location=e.Location.wrap(t),this.history.replace(this.location,this.restorationIdentifier)},r.prototype.historyPoppedToLocationWithRestorationIdentifier=function(t,r){var n;return this.restorationIdentifier=r,this.enabled?(n=this.getRestorationDataForIdentifier(this.restorationIdentifier),this.startVisit(t,"restore",{restorationIdentifier:this.restorationIdentifier,restorationData:n,historyChanged:!0}),this.location=e.Location.wrap(t)):this.adapter.pageInvalidated()},r.prototype.getCachedSnapshotForLocation=function(t){var e;return null!=(e=this.cache.get(t))?e.clone():void 0},r.prototype.shouldCacheSnapshot=function(){return this.view.getSnapshot().isCacheable();
```
Comment (sdelamo) [inline Micronaut/src/main/resources/assets/turbolinks-5.2.0.js:5]: why did you delete it?

**Review by sdelamo — CHANGES_REQUESTED** [review]

File: Micronaut/src/main/resources/assets/turbolinks-5.2.0.js (hunk 8 above)
Comment (timyates) [reply Micronaut/src/main/resources/assets/turbolinks-5.2.0.js:5]: I replaced it with v7

File: Micronaut/src/main/kotlin/io/micronaut/turbo/basic/NonExistentController.kt (hunk 1 above)
Comment (timyates) [reply Micronaut/src/main/kotlin/io/micronaut/turbo/basic/NonExistentController.kt:14]: In their app, they just return a 404 and a Sting, so I copied that

https://github.com/hotwired/turbo-native-demo/blob/fbd6cb6e07eb4fe1661d6330a377237d6ce606a8/server.js#L117-L119

File: Micronaut/src/main/kotlin/io/micronaut/turbo/basic/OneController.kt (hunk 2 above)
Comment (timyates) [reply Micronaut/src/main/kotlin/io/micronaut/turbo/basic/OneController.kt]: 👍
